src/lib/storage/base_segment.hpp

```diff
@@ -32,6 +34,11 @@ class BaseSegment : private Noncopyable {
   // such as strings who memory usage is implementation defined
   virtual size_t estimate_memory_usage() const = 0;
 
+  SegmentAccessStatistics_T& access_statistics() const;
+
+ protected:
+  mutable SegmentAccessStatistics_T _access_statistics;
+
```

Comment:
What do we need the accessor for; why can't the statistics be public. Also, `_T` violates the code style.

Reply:
Didn't even think about that.

Reply:
I don't want anybody to be able to change th counter externally.